### ISSUE TITLE
React-native-map shows empty map when zooming in for some locations

### ISSUE THREAD

**Issue by gitforsubhransu** [react-native-community/react-native-maps#3002]
# Bug
It may sound very silly but RNM framework behaves differently for different locations. The react-native-map works great for California and some other states. But when I try to search for a location in NY or Michigan (and some other places) it just acts weird:

The map zooms in to a random empty location.
When you try to zoom in it zooms out automatically.
Double tap zooms out of the map rather than zooming in.

React native info output:

```bash
    Binaries:
      Node: 10.15.3 - /usr/local/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
      Android SDK:
        API Levels: 26, 27, 29
        Build Tools: 27.0.3, 28.0.3, 29.0.0
        System Images: android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.1 => 16.6.1 
      react-native: 0.57.7 => 0.57.7 
    npmGlobalPackages:
      create-react-native-app: 2.0.2
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
      react-native: 0.59.9

```

Library version: x.x.x

## Steps To Reproduce

1.
2.
...

Describe what you expected to happen:

1.
2.

**Comment by gitforsubhransu** [comment]
Note: I am running it on iOS with default provider (Apple Maps).

**Comment by rborn** [comment]
What's an "empty" location? How does it look like? Can you make some screenshot?

**Comment by gitforsubhransu** [comment]
Hi @rborn , thank you for replying. I have added a screenshot. 


[
<img width="558" alt="Screen Shot 2019-08-02 at 1 47 44 PM" src="https://user-images.githubusercontent.com/53545017/62567869-efacae80-b840-11e9-9dd2-6f919aa0301c.png">
](url)

**Comment by clemenskroell** [comment]
Having the same issue… Any advice?

**Comment by gitforsubhransu** [comment]
I got it to work with a workaround. The problem exist when you update the region with current state of region. I used to change region by updating its coordinates and deltas. That causes abnormal behavior in map(this might be a bug in react-native-map or it may be by design). When I change it to following the map works perfect.

 onRegionChangeComplete={this.onRegionChangeComplete}


onRegionChangeComplete=(region) => {
    console.warn('onRegionChange')
    this.setState({ 
      region: region
     });
  }

Hope it will be helpful to others.
